### ISSUE TITLE
Multiple Losses

### ISSUE THREAD

**Issue by zbl929** [open-mmlab/mmcv#2913]
### What is the feature?

I want to know what is the difference between the following two cases when using the Multiple Losses?

1：
model = dict(
    decode_head=dict(loss_decode=[dict(type='CrossEntropyLoss', loss_name='loss_ce', loss_weight=1.0),
            dict(type='DiceLoss', loss_name='loss_dice', loss_weight=3.0)]),
    auxiliary_head=dict(loss_decode=[dict(type='CrossEntropyLoss', loss_name='loss_ce',loss_weight=1.0),
            dict(type='DiceLoss', loss_name='loss_dice', loss_weight=3.0)]),
    )
2：model = dict(
    decode_head=dict(loss_decode=[
            dict(
                type='CrossEntropyLoss',
                loss_name='loss_ce',
                loss_weight=3.0,
                class_weight=[0.8373, 1.555]
            ),
            dict(
                type='FocalLoss',
                loss_name='loss_focal',
                loss_weight=3.0,
                class_weight=[0.8373, 1.555]
            ),

**Comment by GOKULNATHCOSMOS** [comment]
> ### What is the feature?
> I want to know what is the difference between the following two cases when using the Multiple Losses?
> 
> 1： model = dict( decode_head=dict(loss_decode=[dict(type='CrossEntropyLoss', loss_name='loss_ce', loss_weight=1.0), dict(type='DiceLoss', loss_name='loss_dice', loss_weight=3.0)]), auxiliary_head=dict(loss_decode=[dict(type='CrossEntropyLoss', loss_name='loss_ce',loss_weight=1.0), dict(type='DiceLoss', loss_name='loss_dice', loss_weight=3.0)]), ) 2：model = dict( decode_head=dict(loss_decode=[ dict( type='CrossEntropyLoss', loss_name='loss_ce', loss_weight=3.0, class_weight=[0.8373, 1.555] ), dict( type='FocalLoss', loss_name='loss_focal', loss_weight=3.0, class_weight=[0.8373, 1.555] ),
> 
> ### Any other context?
> _No response_

In both of these cases, you're configuring the loss functions for training a neural network, likely for a semantic segmentation task. The differences between the two cases involve the types of loss functions being used and how they're weighted.

**Case 1**:

In this case, you're using a combination of two loss functions for each decode head (main segmentation task) and auxiliary head (if present). The loss functions are `'CrossEntropyLoss'` and `'DiceLoss'`. The `'CrossEntropyLoss'` is a common choice for segmentation tasks, and the `'DiceLoss'` is another loss function that is often used for measuring the similarity between predicted and target masks.

The `loss_weight` parameter specifies the weight given to each loss function during optimization. The higher the value, the more the network will be trained to minimize that particular loss. In this case, the `'CrossEntropyLoss'` is weighted with `1.0`, and the `'DiceLoss'` is weighted with `3.0`. This means that the network will focus more on minimizing the `'DiceLoss'` compared to the `'CrossEntropyLoss'`.

**Case 2**:

In this case, you're using different loss functions directly within the `loss_decode` list. You're using `'CrossEntropyLoss'` and `'FocalLoss'`. `'FocalLoss'` is another popular loss function used for tasks where there is class imbalance, like in semantic segmentation. It gives more weight to hard-to-classify samples.

In this case, you're also providing `class_weight` lists to both loss functions. This is likely for addressing class imbalance in the dataset. The values in the `class_weight` list give more importance to certain classes during loss computation. In this example, class `0` is given a weight of `0.8373`, and class `1` is given a weight of `1.555`. This means that the loss function will give more emphasis to class `1` during optimization.

**Key Differences**:

- **Loss Functions**: The first case uses `'CrossEntropyLoss'` and `'DiceLoss'`, while the second case uses `'CrossEntropyLoss'` and `'FocalLoss'`.
- **Weights**: In the first case, `'DiceLoss'` is weighted more heavily. In the second case, both `'CrossEntropyLoss'` and `'FocalLoss'` are given the same weight.
- **Class Weights**: The second case incorporates class weights (`class_weight`) to address class imbalance, while the first case doesn't show the usage of class weights in your provided code.

In both cases, the choice of loss functions and their configuration aims to improve the network's ability to learn and make accurate predictions in a semantic segmentation task. The specific choice of loss functions and weights often involves experimentation to find the best configuration for the specific problem and dataset.
